tests/Tests.elm

```diff
@@ -99,6 +99,10 @@ escapeTest =
             \() ->
                 Escape.sanitize " inva_lid st-ring__name--some-end "
                     |> Expect.equal "inva_lidst-ringnamesome-end"
+        , test "sanitize - complex example" <|
+            \() ->
+                Escape.sanitize "  98 hel ___ žščř--ďťň  lo"
+                    |> Expect.equal "hello"
         , test "CaPITAL to lowercase" <|
             \() ->
                 Escape.sanitize "CaPITAL"
```

Comment:
I think this might be a very nice case using Fuzzers, generating different kind of inputs.
https://package.elm-lang.org/packages/elm-explorations/test/latest/Fuzz

Reply:
Good idea, I'll try to add couple of fuzz tests.

Reply:
I was thinking about it too, but you still need to define what is valid and what is not valid so you can go simply with the almost all chars string like
```
" !"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~ ¡¢£¤¥¦§¨©ª«¬­®¯°±²³´µ¶·¸¹º»¼½¾¿ÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖ×ØÙÚÛÜÝÞßàáâãäåæçèéêëìíîïðñòóôõö÷øùúûüýþÿĀ"
```
:)

Reply:
And maybe even better to cover case with valid numbers in middle or in the end.
```
" !"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz123{|}~ ¡¢£¤¥¦§¨©ª«¬­®¯°±²³´µ¶·¸¹º»¼½¾¿ÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖ×ØÙÚÛÜÝÞßàáâãäåæçèéêëìíîïðñòóôõö÷øùúûüýþÿĀ666"
```

Reply:
I was thinking about something like this, counting the substrings of the namespace:

(This will not work, just some ideas and pseudo-code)

```elm
"xx__xx--xx---x--"
    |> sanitize
    |> splitBy "-"
    |> length

isEqual

"xx__xx--xx---x--"
    |> splitBy "-"
    |> length
```

Reply:
Another idea:

```elm
"xx__xx--xx---x--"
    |> sanitize
    |> splitBy "-"
    |> length

isEqual

"xx__xx--xx---x--"
    |> splitBy "-"
    |> List.map sanitize
    |> length
```

Reply:
I wouldn't complicate it too much. Just added sanity fuzz test checking that only valid chars are kept and that sanitized output doesn't start with a digit